### ISSUE TITLE
BEEFY: Simplify hashing for `pallet-beefy-mmr`

### ISSUE THREAD

**Issue by serban300** [paritytech/substrate#12387]
Right now `pallet-beefy-mmr` Declares a new [`Hasher` trait](https://github.com/paritytech/substrate/blob/61b9a4d1a8a9bf39c1d89a8dd02f82785c10860c/frame/beefy-mmr/primitives/src/lib.rs#L51) and a new [`Keccak256` struct](https://github.com/paritytech/substrate/blob/61b9a4d1a8a9bf39c1d89a8dd02f82785c10860c/frame/beefy-mmr/primitives/src/lib.rs#L61) which implements it. I was wondering if this is needed.

I was thinking of 2 possible simplifications:

1. Instead of defining a new struct we could reuse the `Keccak256` struct that's already present in [`sp_runtime`](https://github.com/paritytech/substrate/blob/61b9a4d1a8a9bf39c1d89a8dd02f82785c10860c/primitives/runtime/src/traits.rs#L714), doing something like:
```
#[cfg(feature = "keccak")]
mod keccak256 {
	pub use sp_runtime::traits::Keccak256;

	/// Keccak256 hasher implementation.
	impl super::Hasher for Keccak256 {
		fn hash(data: &[u8]) -> super::Hash {
			Self::hash(data).into()
		}
	}
}
```

Or we could do even something more generic:
```
	impl<T> super::Hasher for T
		where T: sp_runtime::traits::Hash,
		<T as sp_runtime::traits::Hash>::Output: Into<super::Hash> {
		fn hash(data: &[u8]) -> super::Hash {
			Self::hash(data).into()
		}
	}
```

2. Could we go one step further and remove the `Hasher` trait completely and use `sp_runtime::traits::Hash` instead ? And maybe in the end we could also remove [this](https://github.com/paritytech/substrate/blob/61b9a4d1a8a9bf39c1d89a8dd02f82785c10860c/frame/beefy-mmr/src/lib.rs#L166).

@acatangiu WDYT ? If this issue makes sense I would be happy to implement it.

**Comment by acatangiu** [comment]
No 1 sounds good to me, for no.2 it'd be better to see an RFC/PR 👍

**Comment by serban300** [comment]
Implemented in #12393
